src/main.rs

```diff
@@ -1,29 +1,69 @@
-use std::hash::Hash;
-use std::io::ErrorKind;
-use std::iter::FromIterator;
-use std::{
-    collections::HashSet,
-    fs::{self, File},
-    process::Command,
-};
-
+mod handler;
 use anyhow::{bail, Error, Result};
 use clap::{Parser, Subcommand, ValueEnum};
+use config::{Config, File, FileFormat};
 use glob::glob;
-use serde::{Deserialize, Serialize};
+use handler::*;
+use serde::Deserialize;
+use std::path::Path;
+use std::process::Command;
+use std::str;
+
+/// dir that greenboot looks for the health check and other scripts
+static GREENBOOT_INSTALL_PATHS: [&str; 2] = ["/usr/lib/greenboot", "/etc/greenboot"];
+
+/// greenboot config path
+static GREENBOOT_CONFIG_FILE: &str = "/etc/greenboot/greenboot.conf";
 
 #[derive(Parser)]
 #[clap(author, version, about, long_about = None)]
 #[clap(propagate_version = true)]
+
+/// cli parameters for greenboot
 struct Cli {
     #[clap(value_enum, short, long, default_value_t = LogLevel::Info)]
     log_level: LogLevel,
-
     #[clap(subcommand)]
     command: Commands,
 }
+#[derive(Debug, Deserialize)]
+///config params for greenboot
+struct GreenbootConfig {
+    max_reboot: i32,
+}
+
+impl GreenbootConfig {
+    /// sets the default parameter for greenboot config
+    fn set_default() -> Self {
+        Self { max_reboot: 3 }
+    }
+    /// gets the config from the config file
+    fn get_config() -> Self {
+        let mut config = Self::set_default();
+        let parsed = Config::builder()
+            .add_source(File::new(GREENBOOT_CONFIG_FILE, FileFormat::Ini))
+            .build();
+        match parsed {
+            Ok(c) => {
+                config.max_reboot = match c.get_int("GREENBOOT_MAX_BOOT_ATTEMPTS") {
+                    Ok(c) => c.try_into().unwrap_or_else(|e| {
+                        log::warn!("config format error:{e}, using default value");
```

Comment:
Elsewhere we have "{e} log message text", we need to be consistent for audit platforms so we always want the error number first, then the human readable message.